scripts/spack/packages/serac/package.py

```diff
@@ -47,6 +47,8 @@ class Serac(CachedCMakePackage, CudaPackage):
             description='Enable runtime safety and debug checks')
     variant('shared',   default=False,
             description='Enable build of shared libraries')
+    variant('asan', default=False,
```

Comment:
Can you add a conflict if the compiler is not clang or gcc?

Reply:
Good catch! Totally forgot about that restriction.

Reply:
Added checks in [`8916e60f`](https://github.com/LLNL/serac/pull/571/commits/8916e60f5852ee7247e62886bb762fcbebc2d9bc). The checks might be overkill -- comments explain the approach.